### PR TITLE
docs: Fixed init api docs

### PR DESCRIPTION
**Pull request by iwoplaza**

### CODE REVIEW

**Review by reczkok — APPROVED** [review]
🛠️⚙️

File: apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx [hunk 1]
```diff
@@ -1,133 +1,241 @@
...
+```
+Matching WGSL statement:
+```wgsl
+// shader.wgsl
+@group(...) @binding(0) var<uniform> luckyNumber: f32;
+// ...
+```
 
-## TypeGPU approach
+#### Resources that fit this layout entry
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx]: none?

File: apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx [hunk 2]
```diff
@@ -1,133 +1,241 @@
...
-TypeGPU addresses these issues by referencing resources by named keys rather than indices and by providing static type checking when populating bind groups.
-We also provide utilities which can generate those definitions from existing `.wgsl` shaders. This means that you can benefit from the type safety and convenience of TypeGPU without having to worry about keeping the definitions in sync with the shader code.
 
-### Typed Bind Group Layouts
+### Storage
 
-First, let's cover how to manually create a typed bind group layout in TypeGPU.
-To create a typed bind group layout, you need to call `tgpu.bindGroupLayout` and pass an object that describes the resources that the bind group should contain.
-Here is an example of how you can create a typed bind group layout that matches the previous example:
+To get readonly/mutable access to a buffer, create a property of with the value matching:
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx]:
```suggestion
To get readonly/mutable access to a buffer, create a property with the value matching:
```

File: apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx [hunk 3]
```diff
@@ -1,133 +1,241 @@
...
+```ts
+{
+  sampler: 'filtering' | 'non-filtering' | 'comparison';
+}
+```
 
-Here is an example of how you can create a typed bind group that matches the previous example:
+### Textures
 
+To be able to sample a texture in a shader, create a property of with the value matching:
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx]:
```suggestion
To be able to sample a texture in a shader, create a property with the value matching:
```

File: apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx [hunk 4]
```diff
@@ -1,133 +1,241 @@
...
+  values: array<f32>;
+}
+
+@group(...) @binding(0) var<storage, read> factors: array<f32>;
+@group(...) @binding(1) var<storage, read> filter: Filter;
+// ...
+```
 
-### Typed Bind Groups
+### Samplers
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx:154]: self-explanatory I suppose?

File: apps/typegpu-docs/src/content/docs/fundamentals/buffers.mdx [hunk 5]
```diff
@@ -192,248 +97,106 @@ You can also add all flags in a single `$usage()`.
...
+// .write(data: { position: vec2f, health: number })
+particleBuffer.write({
+  position: vec2f(1.0, 2.0),
+  health: 100,
+});
+```
 
-### Full migration
+:::caution[WebGPU Interoperability]
+If you create a buffer from an existing WebGPU buffer than happens to be mapped, the data will be written directly to the buffer (the buffer will not be unmapped).
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/buffers.mdx]:
```suggestion
If you create a buffer from an existing WebGPU buffer that happens to be mapped, the data will be written directly to the buffer (the buffer will not be unmapped).
```

File: apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx (hunk 4 above)
Comment (iwoplaza) [reply apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx:154]: I wrote a bit more text upfront, does it look better now? 👀

Comment (mhawryluk) [reply apps/typegpu-docs/src/content/docs/fundamentals/bind-groups.mdx:154]: yup!

**Review by mhawryluk — APPROVED** [review]
Awesome 🚀

**Review by reczkok — APPROVED** [review]
Great work!📖🧑‍🏫

**Pull request merged**